lib/Sema/CSSolver.cpp

```diff
@@ -1201,14 +1209,15 @@ void ConstraintSystem::shrink(Expr *expr) {
     }
 
     bool isSuitableCollection(TypeRepr *collectionTypeRepr) {
-      // Only generic identifier, array or dictionary.
+      // Only generic identifier, array, dictionary, and inline array.
       switch (collectionTypeRepr->getKind()) {
       case TypeReprKind::UnqualifiedIdent:
         return cast<UnqualifiedIdentTypeRepr>(collectionTypeRepr)
             ->hasGenericArgList();
 
       case TypeReprKind::Array:
       case TypeReprKind::Dictionary:
+      case TypeReprKind::InlineArray:
```

Comment:
At this point I think we shouldn't expand shrink to new types, let's let the proper solver handle it...

Reply:
Fair enough, should we also decline to handle the non-sugared version?